connection.go

```diff
@@ -561,8 +561,11 @@ func (sc *snowflakeConn) WaitForQueryCompletion(ctx context.Context, qid string)
 		return nil
 	}
 	// query is complete because of an error; return that error
-	if err.(*SnowflakeError).Number != ErrQueryIsRunning {
-		return err
+	switch v := err.(type) {
+	case *SnowflakeError:
+		if v.Number != ErrQueryIsRunning {
+			return err
+		}
```

Comment:
What is happening in all the other cases ? Is the error going to dissapear?

Reply:
Oops I switched the logic there. Now it should be correct... if the error is running, block waiting for it to complete or else return the error